InteractiveHtmlBom/web/render.js

```diff
@@ -419,20 +419,41 @@ function drawTracks(canvas, layer, defaultColor, highlight) {
   ctx.lineCap = "round";
   for (var track of pcbdata.tracks[layer]) {
     if (highlight && highlightedNet != track.net) continue;
-    ctx.strokeStyle = highlight ? defaultColor : settings.netColors[track.net] || defaultColor;
-    ctx.lineWidth = track.width;
-    ctx.beginPath();
-    if ('radius' in track) {
-      ctx.arc(
-        ...track.center,
-        track.radius,
-        deg2rad(track.startangle),
-        deg2rad(track.endangle));
+    if ('drillsize' in track && track.start[0] == track.end[0] && track.start[1] == track.end[1]) {
+      var pad = {
+        type: "th",
+        pos: track.start,
+        layers: ['F', 'B'],
+        net: track.net,
+        angle: 0,
+        shape: "circle",
+        size: [track.width, track.width],
+        drillshape: "circle",
+        drillsize: [track.drillsize, track.drillsize]
+      }
+      var style = getComputedStyle(topmostdiv);
+      var colors = {
+        pad: highlight ? style.getPropertyValue('--pad-color-highlight') : style.getPropertyValue('--pad-color'),
+        padHole: style.getPropertyValue('--pad-hole-color')
+      }
+      drawPad(ctx, pad, colors.pad, false)
+      drawPadHole(ctx, pad, colors.padHole)
```

Comment:
This is way more complex than it needs to be and will slow down rendering.
Just calculate colors and then do
```
ctx.strokeStyle = copperColor;
ctx.lineWidth = track.width;
ctx.moveTo(...track.start);
ctx.lineTo(...track.end);
ctx.strokeStyle = holeColor;
ctx.lineWidth = track.drillsize;
ctx.lineTo(...track.end);
```
this will only draw 2 circles and we don't have to deal with the complexities of the pad.